src/runtime/link-accounts-flow.ts

```diff
@@ -266,57 +259,42 @@ export class LinkCompleteFlow {
  * linked.
  */
 export class LinkSaveFlow {
-  /**
-   * @param {!./deps.Deps} deps
-   * @param {!../api/subscriptions.SaveSubscriptionRequestCallback} callback
-   */
-  constructor(deps, callback) {
-    /** @private @const {!Window} */
-    this.win_ = deps.win();
+  private readonly win_: Window;
+  private readonly activityPorts_: ActivityPorts;
+  private readonly dialogManager_: DialogManager;
 
-    /** @private @const {!./deps.Deps} */
-    this.deps_ = deps;
+  private requestPromise_: Promise<SaveSubscriptionRequest | void> | null =
+    null;
+  private activityIframeView_: ActivityIframeView | null = null;
 
-    /** @private @const {!../components/activities.ActivityPorts} */
-    this.activityPorts_ = deps.activities();
+  /** Visible for testing. */
+  openPromise: Promise<void> | null = null;
 
-    /** @private @const {!../components/dialog-manager.DialogManager} */
-    this.dialogManager_ = deps.dialogManager();
+  constructor(
+    private readonly deps_: Deps,
+    private readonly callback_: SaveSubscriptionRequestCallback
+  ) {
+    this.win_ = deps_.win();
 
-    /** @private {!../api/subscriptions.SaveSubscriptionRequestCallback} */
-    this.callback_ = callback;
+    this.activityPorts_ = deps_.activities();
 
-    /** @private {?Promise<!../api/subscriptions.SaveSubscriptionRequest>} */
-    this.requestPromise_ = null;
-
-    /** @private {?Promise} */
-    this.openPromise_ = null;
-
-    /** @private {?ActivityIframeView} */
-    this.activityIframeView_ = null;
+    this.dialogManager_ = deps_.dialogManager();
   }
 
   /**
-   * @return {?Promise<!../api/subscriptions.SaveSubscriptionRequest>}
-   * @package Visible for testing.
+   * Visible for testing.
    */
-  getRequestPromise() {
+  getRequestPromise(): Promise<SaveSubscriptionRequest | void> | null {
     return this.requestPromise_;
   }
 
-  /**
-   * @private
-   */
-  complete_() {
+  private complete_(): void {
     this.dialogManager_.completeView(this.activityIframeView_);
   }
 
-  /**
-   * @param {!Object} result
-   * @return {!Promise<boolean>}
-   * @private
-   */
-  async handleLinkSaveResponse_(result) {
+  private async handleLinkSaveResponse_(
+    result: LinkCompleteResponse
```

Comment:
is this intended? should result be of type `{}`?

Reply:
I just improved the types

Reply:
Now there are types for `LinkSaveResponse`, `LinkConfirmResponse`, and `LinkCompleteResponse` (a combination of the first two)

Reply:
nice!